### PR TITLE
Build operator dependencies first before testing it

### LINKED ISSUE #11641: Keycloak Operator CI build doesn't build Keycloak common
### Describe the bug

When operator-ci.yml runs `mvn clean verify` for the operator, it uses the keycloak-common module from SNAPSHOT that is present in the cache. This is not necessarily the keycloak-common module present in the sources to be tested.

### Version

999-SNAPSHOT

### Expected behavior

Use the branch's source code to build the dependencies.

### Actual behavior

Uses some other version. Saw this in an action run here: https://github.com/keycloak/keycloak/runs/6153105015?check_suite_focus=true

### PR DESCRIPTION
**Pull request by ahus1**
Closes #11641

### CODE REVIEW

**Comment by andreaTP** [comment]
Hi @ahus1 ! And thanks a lot for opening this issue and proposing a fix!

Other than explicitly building the `common` module I would suggest that we change the `mvn clean verify` command to use the `-am` option, something like:
```bash
mvn -P operator -pl operator -am clean verify \
    ....
```

This way we do not need to track dependencies in the future.

**Comment by ahus1** [comment]
@andreaTP - I wasn't sure how much time test tests of keycloak-common would take; testing it locally it is 1 sec. Looking at that result, your solution is a lot more robust. I pushed a change, let's see how the builds run with that change. I'll ask you for a review once the build completed.

**Comment by ahus1** [comment]
@andreaTP - the builds are all-green (except the quickstarts). Ready for a review!

**Review by andreaTP — CHANGES_REQUESTED** [review]
LGTM after removing the old implementation of it 🙂 
Thanks a ton!

File: .github/workflows/operator-ci.yml
Comment (ahus1) [reply .github/workflows/operator-ci.yml]: Sorry, you're right. Pushed the update.

**Review by andreaTP — APPROVED** [review]

**Comment by andreaTP** [comment]
I'm extremely sad that we haven't merged this simple and effective PR for so long.

Unfortunately, we are now at the point where another PR ( #11929 ) is touching almost the same bits.

@ahus1 would you mind verifying? And we can probably close this one 😞

**Comment by ahus1** [comment]
@andreaTP - I see that the other PR is touch quite some bits and pieces, and I would need to build it locally to see if it tackes the same issues.

Once that other PR has been merged, I'm happy to check if this PR is still necessary and will then either rebase or close it.

**Review by pedroigor — APPROVED** [review]

**Pull request merged**
